renderer.php

```diff
@@ -110,4 +114,107 @@ public function render_notif_message_teacher($feedback, $courseid) {
         $message .= ' | ' . $this->render_results_link($courseid, $feedback->id);
         return $message;
     }
+
+    /**
+     * Renders the essayquestions
+     *
+     * @param array $essayquestions - array of question objects $essayquestions
+     * @param int $courseid
+     * @param int $feedbackid
+     * @return string
+     */
+    public function render_essay_questions($essayquestions, $courseid, $feedbackid) {
+
+        // Provide downloadlink for textanswers for all questions.
+        $params = [
+            "course" => $courseid,
+            "feedback" => $feedbackid,
+            "download" => "csv"
+        ];
+        $essayhtml = html_writer::link(new moodle_url("/blocks/coursefeedback/essayanswers.php", $params),
+            get_string("page_link_download", "block_coursefeedback", "CSV"));
+
+        // Create and fill Textanswer table.
+        $table = new html_table();
+        $table->data = [];
+
+        foreach ($essayquestions as $essayquestion) {
+            // Cell 1: "Question <id>:"
+            $text = get_string("form_header_question", "block_coursefeedback", $essayquestion->questionid)
+                . ": ";
+
+            // Cell 2: "<questiontext>"
+            $questiontext = format_string($essayquestion->question);
+
+            //Cell 3 "Show answers" (lnk).
+            $params = [ 'course' => $courseid,
+                    'feedback' => $essayquestion->coursefeedbackid,
+                    'question' => $essayquestion->questionid ];
+            $answerlink = html_writer::link(new moodle_url("/blocks/coursefeedback/essayanswers.php", $params),
+                get_string("table_html_showanswers", "block_coursefeedback"));
+
+            // Insert in table
+            $cells = [ $text, $questiontext, $answerlink ];
+            $row = new html_table_row($cells);
+            $table->data[] = $row;
+
+        }
+        $essayhtml .= html_writer::table($table);
+        return $essayhtml; //$html;
```

Comment:
Der Kommentar scheint ein Überbleibsel zu sein.

Reply:
Entfernt in: [da24975](https://github.com/eberhardt/moodle-block_coursefeedback/pull/42/commits/da24975952ae3f59893485e94af2ad72433b1a69)